datasource/utils.go

```diff
@@ -4,6 +4,49 @@ package datasource
 
 import "strconv"
 
+// Any crseAttrValue's that don't fall into these are labeled "None"
+var divisionalAreas = [3]string{"A&H", "SCI", "SOC"}
+
+func getGenEdInfo(intAttributes []any) []string {
+
+	// Get the gen-ed areas from the course attributes.
+	areas := []string{}
+	for _, item := range intAttributes {
+		attrMap := item.(map[string]any)
+		if attrMap["crseAttribute"] == "LGE" {
+			areas = append(areas, attrMap["crseAttrValue"].(string))
+		}
+	}
+	return areas // if empty, not a GENED.
+}
+
+func checkDivisionalArea(val string) bool {
+	for _, area := range divisionalAreas {
+		if val == area {
+			return true
+		}
+	}
+	return false
+}
+
+func getDivisionalInfo(intAttributes []interface{}) []string {
+
+	// Type conversion, since we're dealing with an interface.
+	areas := []string{}
+	for _, item := range intAttributes {
+		attrMap, _ := item.(map[string]any)
+		divAttr, ok := attrMap["crseAttrValue"].(string)
+		if !ok {
+			continue
+		} // sometimes the value is nil. Just move to the next one.
+
+		if attrMap["crseAttribute"] == "LDD" && checkDivisionalArea(divAttr) {
+			areas = append(areas, divAttr)
+		}
```

Comment:
I think it's a matter of personal preference, but IMO you could more directly do what your comment says: Check whether `divAttr` is `nil` directly rather than checking the side effect.
```suggestion
		if maybeDivAttr := attrMap["crseAttrValue"]; maybeDivAttr == nil {
			// Sometimes the value is nil. Just move to the next one.
			continue
		} else if divAttr := maybeDivAttr.(string); attrMap["crseAttribute"] == "LDD" && checkDivisionalArea(divAttr) {
			areas = append(areas, divAttr)
		}
```
(The `else` isn't necessary; just thought it was cleaner. Anyway do whatever. 😛)

Reply:
Thanks for the suggestion! This is cleaner, but I think it makes the code harder to read—I will stick with the original :)